core/dbt/task/retry.py

```diff
@@ -57,59 +62,61 @@
 
 class RetryTask(ConfiguredTask):
     def __init__(self, args, config, manifest) -> None:
-        super().__init__(args, config, manifest)
-
-        state_path = self.args.state or self.config.target_path
-
-        if self.args.warn_error:
-            RETRYABLE_STATUSES.add(NodeStatus.Warn)
-
-        self.previous_state = PreviousState(
```

Comment:
We do not need the full previous state here, just run results.
Loading the whole state could be memory intensive for large projects(manifest can get large.)